docs/api/rtos/Thread.md

```diff
@@ -89,6 +89,12 @@ The Callback API provides a convenient way to pass arguments to spawned threads.
 
 [![View code](https://www.mbed.com/embed/?url=https://os.mbed.com/teams/mbed_example/code/rtos_threading_with_callback/)](https://os.mbed.com/teams/mbed_example/code/rtos_threading_with_callback/file/5938bdb7b0bb/main.cpp)
 
+## Debugging tips
+
+When debugging Thread, check the full RTX state (ready, blocked and so on) for a deadlock (two things waiting for each other) or a missed signal leading to an event machine stall. You can confirm a deadlock if you see two threads in the "blocked" state waiting for a resource that the other is supposed to signal.
+
+To reduce deadlocks, proactively code in a safe way by maintaining a mental ordering of your mutexes and never claiming a high-level mutex while holding a low-level one.
```

Comment:
"mental ordering" sounds weird. Maybe just "ordering"?

I guess the point is that the system doesn't enforce this, so there's no way to follow this rule apart from self-discipline. Feels like you needs some sort of adjective to suggest that, but can't think of anything better.

(For comparison, in Symbian, to help with this, every mutex had a program-specified level number, and debug builds would stop with a diagnostic if you ever claimed a high-level one while holding a low-level one.  Although being a debug-build-only check was annoying - if someone broke the rule and never tried a debug build, then other people found out when they made a debug build to look at their stuff and found the system stopped with a mutex diagnostic in debug mode. Debug builds weren't tested in CI.)